module/input/NatNet/src/NatNet.cpp

```diff
@@ -104,9 +106,14 @@ namespace module::input {
             // Work out our remotes address
             sockaddr_in address{};
             memset(&address, 0, sizeof(sockaddr_in));
-            address.sin_family      = AF_INET;
-            address.sin_port        = htons(cfg.command_port);
-            address.sin_addr.s_addr = htonl(remote);
+            address.sin_family = AF_INET;
+            address.sin_port   = htons(cfg.command_port);
+
+            // Convert IP address from string to binary form
+            if (inet_pton(AF_INET, remote.c_str(), &address.sin_addr) <= 0) {
+                log<NUClear::ERROR>("Invalid remote IP address");
+                return;
+            }
```

Comment:
Rather than all this and the sendto, you can just emit<UDP> instead then you can just pass the address in without having to do the conversion yourself

Reply:
Or leave it, when I first wrote this I had dreams of making this natnet stuff a separate library so others could use it too